File: backend/src/models/Emergency.ts
import { EmergencyType } from "./EmergencyType.js";

export class Emergency {
    id: number;
    name: string;
    emergencyType: EmergencyType | null;
    
    constructor(id: number, name: string, emergencyType: EmergencyType | null) {
        this.id = id;
        this.name = name;
        this.emergencyType = emergencyType;
    }

    toJSON() {
        return {
            id: this.id,
            name: this.name,
            emergencyType: this.emergencyType ? this.emergencyType.id : null
        };
    }
}

File: lib/models/emergency.dart
/// Mirrors the backend's Emergency entity (id + name).
class Emergency {
  final int id;
  final String name;
  final int emergencyTypeId;

  const Emergency({required this.id, required this.name, required this.emergencyTypeId});

  factory Emergency.fromJson(Map<String, dynamic> json) {
    return Emergency(
      id: json['id'] as int,
      name: json['name'] as String,
      emergencyTypeId: json['emergencyType'] as int
    );
  }
}
